pageserver/src/tenant/timeline.rs

```diff
@@ -2874,6 +2889,16 @@ impl Timeline {
             .schedule_index_upload_for_gc_compaction_state_update(gc_compaction_state)
     }
 
+    pub(crate) fn update_rel_size_v2_status(
+        &self,
+        rel_size_v2_status: RelSizeMigration,
+    ) -> anyhow::Result<()> {
+        self.rel_size_v2_status
+            .store(Some(Arc::new(rel_size_v2_status.clone())));
```

Comment:
We should update this after the index upload is scheduled, since it can fail.

Also, nit: this is a bare enum, so we can derive `Copy` and avoid the cloning.

Reply:
`schedule_index_upload_for_rel_size_v2_status_update` doesn't guarantee the index gets updated.

Because the layer uploads with reldirv2 are put into the queue after the index_part update of the `rel_size_v2_status`, I believe we won't race in such cases.